runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -320,125 +238,303 @@ private NodeStatus getLogUnitNodeStatusInLayout(Layout layout, String server) {
+                        entry.getKey(), e);
+            }
         }
 
-        RuntimeLayout runtimeLayout = new RuntimeLayout(layout, runtime);
+        // If discoverLayoutServers is set, fetch layouts from all other layout servers
+        // discovered in the first search which are not contained in the initial list.
+        if (discoverLayoutServers) {
+            List<String> discoveredLayoutServers = layoutMap.values().stream()
+                    .map(l -> l.getLayoutServers())
+                    .flatMap(s -> s.stream())
+                    .distinct()
+                    .filter(s -> !layoutServers.contains(s))
+                    .collect(Collectors.toList());
+
+            if (!discoveredLayoutServers.isEmpty()) {
+                Map <String, Layout> recursiveLayouts = getLayouts(new ArrayList<>(discoveredLayoutServers), false);
+                layoutMap.putAll(recursiveLayouts);
+            }
+        }
 
-        // All configured nodes.
-        Set<String> allNodes = new HashSet<>(layout.getAllServers());
+        return layoutMap;
+    }
 
-        // Counters to track heartbeat responses from nodes.
-        // A counter is incremented even if we get a WrongEpochException as the node is alive.
-        Map<String, Integer> counters = new HashMap<>();
-        allNodes.forEach(endpoint -> counters.put(endpoint, 0));
+    private Layout getLayoutFromQuorum(Map<String, Layout> layoutMap, int quorum) {
+        Layout layout = null;
+        Map<Layout, Integer> uniqueLayoutMap = null;
+
+        // Find if layouts are the same in all nodes (at least quorum nodes should agree in the
+        // same layout) - count occurrences of each layout in cluster nodes.
+        for (Entry<String, Layout> entry : layoutMap.entrySet()) {
+            if (uniqueLayoutMap == null) {
+                uniqueLayoutMap = new HashMap<Layout, Integer>() {{
+                    put(entry.getValue(), 1);
+                }};
+            } else {
+                Boolean sameLayoutFound = false;
+                for (Layout uniqueLayout : uniqueLayoutMap.keySet()) {
+                    if (entry.getValue().equals(uniqueLayout)) {
+                        uniqueLayoutMap.merge(uniqueLayout, 1, Integer::sum);
+                        sameLayoutFound = true;
+                        break;
+                    }
+                }
+                if (!sameLayoutFound) {
+                    uniqueLayoutMap.put(entry.getValue(), 1);
+                }
+            }
+        }
+
+        // Filter layouts present in quorum number of nodes
+        Map<Layout, Integer> uniqueLayoutsQuorumMap = uniqueLayoutMap.entrySet().stream()
+                .filter(x -> x.getValue() >= quorum)
+                .collect(Collectors.toMap(x -> x.getKey(), x -> x.getValue()));
+
+        // Select layout with greater number of occurrences.
+        if (uniqueLayoutsQuorumMap.size() > 0) {
+            Map.Entry<Layout, Integer> maxEntry = null;
```

Comment:
It can be simplified to: 
Put all values from the Map to a TreeSet then `getFirst()`